### PR TITLE
Add docs for adding Chainlink Functions to projects

### PR DESCRIPTION
**Pull request by dwightjl**
Add docs explaining how to use Chainlink Functions without starter kit tasks. All examples are included on docs.chain.link except the request script and the request source file, which are pulled from https://github.com/dwightjl/functions-examples

### CODE REVIEW

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+You must have a wallet with testnet LINK and native tokens to run this example. This example uses Polygon Mumbai. You can get testnet LINK at [faucets.chain.link](https://faucets.chain.link/mumbai) and testnet MATIC at the [Polygon Faucet](https://faucet.polygon.technology/). To get testnet funds for other networks, see the [LINK Token Contracts](/resources/link-token-contracts) page.
+
+:::note[Request Access]
+Chainlink Functions is currently in a limited BETA.
+Apply [here](http://functions.chain.link/) to add your EVM account address to the Allow List.
+:::
+
+### Configure Hardhat
+
+If you already have a project, you can skip to the [Libraries and Dependencies](#get-the-libraries-and-contract-dependencies) section. If you don't already have a Hardhat project, start one with the following steps:
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:41]: can we move this disclaimer about 'if you have a project already, skip ahead' further up to be before the configure hardhat section? I feel it belongs there more?

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+# Wallet private key
+PRIVATE_KEY="your_private_key"
+```
+
+Run `npx hardhat compile` to make sure Hardhat is configured correctly. You will see several warnings, but everything is correct if the Solidity files compile successfully.
+
+## Get the dependency contracts and scripts
+
+Get the required libraries from the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit). You can clone the repo and get the folders manually, or use a single command to get a tarball through the GitHub API and extract the folders you need to the correct location in your project:
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:105]: suggest changing 'You can clone the repo and get the folders manually'  to 'You can clone the repo and copy across the folders into your existing project manually'

Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:105]: what folder do they run this git command from? The root folder of their project?

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+tar -xf ../functions.tar.gz --strip=1 --wildcards smartcontractkit-functions-hardhat-starter-kit-*/contracts/dev smartcontractkit-functions-hardhat-starter-kit-*/contracts/test smartcontractkit-functions-hardhat-starter-kit-*/FunctionsSandboxLibrary smartcontractkit-functions-hardhat-starter-kit-*/contracts/FunctionsConsumer.sol
+```
+
+When you are done, you should have the necessary dependencies in the following directories:
+
+- `contracts/dev`
+- `contracts/test`
+- `FunctionsSandboxLibrary`
+
+Run `npx hardhat compile` again to make sure all the dependencies are satisfied. Add missing dependencies or Hardhat configuration options as necessary.
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:118]: or whatever the command to compile all contracts is in your dev env of choice

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+## Configure on-chain resources
+
+The on-chain resources are critical for Chainlink Functions to process your requests.
+
+1. [Create a consumer contract](#create-a-consumer-contract)
+1. [Deploy a consumer contract](#deploy-a-consumer-contract)
+
+### Create a consumer contract
+
+Use the [FunctionsConsumer.sol](https://github.com/smartcontractkit/functions-hardhat-starter-kit/blob/main/contracts/FunctionsConsumer.sol), which is already in your `./contracts` folder. After you complete this tutorial, you can modify it to fit your needs and redeploy it.
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:129]: 'You can use the FunctionsConsumer.sol......you can modify it to fit your needs and redeploy it, or you can use your existing contract and ensure it adheres to all the requirements listed below'

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+
+In general, a consumer contract requires several components:
+
+<ChainlinkFunctions section="functions-consumer" />
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:133]: i see this import in here:
`import "@chainlink/contracts/src/v0.8/dev/functions/FunctionsClient.sol";`

is this right? don't we have to manually reference the library in our project. My understanding is we don't have functions contracts in the NPM lib

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+
+Next, create and fund your Chainlink Functions subscription.
+
+### Create and fund a subscription
+
+You can use the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit) to create and manage your subscriptions. See [Managing Subscriptions](/chainlink-functions/resources/subscriptions) for instructions.
+
+This example shows how to create and manage subscriptions programmatically. You can create the subscription, fund the subscription, and authorize the consumer all in one script.
+
+1. Ensure that the wallet address you are using to create the subscription has a sufficient LINK balance. You can get testnet LINK at [faucets.chain.link](https://faucets.chain.link/mumbai). To get testnet funds for other networks, see the [LINK Token Contracts](/resources/link-token-contracts) page.
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:187]: do we also need the discaimer about being signed up for the whitelist here?

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+- **Secrets:** Optional secrets that your source code needs to access APIs or other interfaces. Secrets can be included inline with your request or hosted off-chain. See the [Using Secrets in Requests](/chainlink-functions/tutorials/api-use-secrets) and [Using Off-chain Secrets](/chainlink-functions/tutorials/api-use-secrets-offchain) tutorials for examples.
+- **Secrets location:** This tells the DON where to obtain your secrets.
+
+### Create a request script
+
+Download some example source code, and create a script to assemble the required components. The script will read your source code, define arguments, encrypt secrets, and send requests to your consumer contract. This example script does not require Hardhat, so you can modify it to run in a browser using Ethers or another framework.
+
+1. Use `curl` to get the example source code. This code runs on each node in the DON and returns a response to your consumer contract. For this example, use the source from the [Call an API](/chainlink-functions/tutorials/api-query-parameters) tutorial. The following `curl` request creates a file named `Functions-request-source.js` with the source code:
```
Comment (pappas999) [inline src/pages/chainlink-functions/resources/add-functions-to-projects.md:245]: do we need to specify here how this is showing an example, but if they have their own JS code, they should just skip this and refernece it in the next steps?

**Review by pappas999 — APPROVED** [review]
added a few suggestions for consideration, but otherwise LGTM

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 6 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md:196]: Added a reminder here and a notice at the top of the page. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 1 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md]: Moved up to the Before you begin section. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 2 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md]: Added. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md]: I added a clarifying step to change directories to project root. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 3 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md]: Added. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 4 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md]: Added. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 5 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md:142]: I corrected this in the import to say that the file must be downloaded manually and imported from within the project. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

File: src/pages/chainlink-functions/resources/add-functions-to-projects.md (hunk 7 above)
Comment (dwightjl) [reply src/pages/chainlink-functions/resources/add-functions-to-projects.md:256]: Good suggestion. I clarified that this is the point where you can include your own source file. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0

**Pull request merged**